apps/backend/src/poll/poll/poll.service.ts

```diff
@@ -126,228 +152,237 @@ export class PollService implements OnModuleInit {
+      if (!oldEvent) {
+        return false;
+      }
+      return oldEvent.start.valueOf() !== event.start.valueOf() || oldEvent.end.valueOf() !== event.end.valueOf();
+    });
+    if (updatedEvents.length > 0) {
+      for (const event of updatedEvents) {
+        await this.pollEventModel.findByIdAndUpdate(event._id, event).exec();
+      }
     }
 
-    async postEvents(poll: Types.ObjectId, pollEvents: PollEventDto[]): Promise<PollEvent[]> {
-        const oldEvents = await this.pollEventModel.find({poll}).exec();
-        const newEvents = pollEvents.filter(event => !oldEvents.some(oldEvent => oldEvent._id.toString() === event._id));
-        await this.pollEventModel.create(newEvents.map(event => ({...event, poll})));
-
-        const updatedEvents = pollEvents.filter(event => {
-            const oldEvent = oldEvents.find(e => e._id.toString() === event._id);
-            if (!oldEvent) {
-                return false;
-            }
-            return oldEvent.start.valueOf() !== event.start.valueOf() || oldEvent.end.valueOf() !== event.end.valueOf();
-        });
-        if (updatedEvents.length > 0) {
-            for (const event of updatedEvents) {
-                await this.pollEventModel.findByIdAndUpdate(event._id, event).exec();
-            }
-        }
+    const deletedEvents = oldEvents.filter(event => !pollEvents.some(e => e._id === event._id.toString()));
+    await this.pollEventModel.deleteMany({_id: {$in: deletedEvents.map(event => event._id)}}).exec();
+    await this.removeParticipations(poll, updatedEvents);
+    return await this.pollEventModel.find({poll}).exec();
+  }
 
-        const deletedEvents = oldEvents.filter(event => !pollEvents.some(e => e._id === event._id.toString()));
-        await this.pollEventModel.deleteMany({_id: {$in: deletedEvents.map(event => event._id)}}).exec();
-        await this.removeParticipations(poll, updatedEvents);
-        return await this.pollEventModel.find({poll}).exec();
+  async getParticipants(id: Types.ObjectId, token: string): Promise<ReadParticipantDto[]> {
+    const poll = await this.pollModel.findById(id).exec();
+    if (!poll) {
+      throw new NotFoundException(id);
     }
+    const currentParticipant = await this.participantModel.find({
+      poll: new Types.ObjectId(id),
+      token,
+    }).exec();
 
-    async getParticipants(id: Types.ObjectId, token: string): Promise<ReadParticipantDto[]> {
-        const participants = await this.participantModel.find({
-            poll: new Types.ObjectId(id),
-            token: {$ne: token},
-        }).select(readParticipantSelect).exec();
-        const currentParticipant = await this.participantModel.find({
-            poll: new Types.ObjectId(id),
-            token,
-        }).exec();
-        return [...participants, ...currentParticipant];
+    if (await this.isAdmin(id, token) || poll.settings.showResult !== ShowResultOptions.NEVER) {
```

Comment:
I think isAdmin re-fetches the poll from the database, but here it is already known.

Reply:
Also missing the condition for AFTER_PARTICIPATING and currentParticipant not empty